src/nvim/ex_getln.c

```diff
@@ -5957,6 +5957,27 @@ char_u *get_cmdline_str(void)
   return vim_strnsave(p->cmdbuff, (size_t)p->cmdlen);
 }
 
+// Set the current command line to a given string.
+// Only works when the command line is being edited.
+// Returns -1 when something is wrong.
+int set_cmdline_str(char *content, size_t len)
+{
+    if (cmdline_star > 0) {
+        return -1;
+    }
+
+    struct cmdline_info *p = get_ccline_ptr();
+
+    if (p == NULL) {
+        return -1;
+    }
+
+    cmdline_del(0);
+    put_on_cmdline((char_u *)content, (int)len, true);
```

Comment:
Does this handle triggering `CmdlineChanged`? From a quick look on mobile I couldn't see the call.

Reply:
Seems to from a quick test, should be able to add that to the actual tests I'm writing though (I think).

Reply:
Actually, this doesn't seem to handle `CmdlineChanged`. But looking at the code, I need a reference to a `CommandLineState` object to call the handler for that? But I'm not sure if there's really a good way for me to get that so . . . not quite sure what to do about this.